assignments/this.js

```diff
@@ -1,26 +1,49 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. When outside of a function, 'this' refers to the window object.
+* 2. The object before the dot on methods is what 'this' refers to.
+* 3. When 'new' is used to create a new object, 'this' refers to that object.
+* 4. When using '.call' or '.apply', 'this' must be explicitly defined.
```

Comment:
using .call / .apply , the first argument is object you want to set THIS equal to.  in the case of constructor functions, when we do a .call(this), that is referring to the hidden object created by the new keyword.